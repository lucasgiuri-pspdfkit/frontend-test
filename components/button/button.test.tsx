import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./button";

const availableTypes = ["button", "submit", "reset"];

describe("Button", () => {
  availableTypes.forEach((type) => {
    it(`Snapshot with type ${type}`, () => {
      const onClick = jest.fn();
      const { container } = render(
        <Button text="test" type={type} onClick={onClick} />
      );
      expect(container).toMatchSnapshot();
    });
  });

  it("handles onClick event ", () => {
    const onClick = jest.fn();
    render(
      <Button text="test" type="button" onClick={onClick} />
    );
    fireEvent.click(screen.getByText("test"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
