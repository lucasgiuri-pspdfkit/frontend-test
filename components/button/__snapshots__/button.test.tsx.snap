// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot with type button 1`] = `
<div>
  <button
    class="px-4 py-2 font-semibold bg-gray-50 text-gray-450 rounded"
    type="button"
  >
    test
  </button>
</div>
`;

exports[`Button Snapshot with type reset 1`] = `
<div>
  <button
    class="px-4 py-2 font-semibold bg-gray-50 text-gray-450 rounded"
    type="reset"
  >
    test
  </button>
</div>
`;

exports[`Button Snapshot with type submit 1`] = `
<div>
  <button
    class="px-4 py-2 font-semibold bg-gray-50 text-gray-450 rounded"
    type="submit"
  >
    test
  </button>
</div>
`;
